nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_prototype.rs

```diff
@@ -732,8 +734,87 @@ impl StringPrototype {
         Ok(String::from_str(agent, substring).into_value())
     }
 
-    fn split(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    /// ### [22.1.3.23 String.prototype.split ( separator, limit )](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.split)
+    fn split(agent: &mut Agent, this_value: Value, args: ArgumentsList) -> JsResult<Value> {
+        // 1. Let O be ? RequireObjectCoercible(this value).
+        let o = require_object_coercible(agent, this_value)?;
+
+        // 2. If separator is neither undefined nor null, then
+        let separator = args.get(0);
+
+        if matches!(separator, Value::Undefined | Value::Null) {
+            // TODO: get_method(separator, Symbol.split)
```

Comment:
**issue:** This shouldn't be too hard: `get_method` function already exists, and then you just perform a `call_function(splitter, separator, Some(&[o, limit]))`

Reply:
I didn't implement this because I don't know how to create the `Symbol.split` value in Rust.
But if you help me with that, I can implement this.

Reply:
Ah, these "well-known symbols" can be "created" as simply as `WellKnownSymbolIndexes::Split.into()` :)

Reply:
Done in the last commit.